tools/cldr-apps/src/org/unicode/cldr/web/SurveyAjax.java

```diff
@@ -2543,4 +2532,199 @@ public String getValue() {
         r.put("phStatus", phStatus);
         r.put("covLev", covLev);
     }
+
+    /**
+     * Get data associated with a particular row (xpath).
+     *
+     * @param request
+     * @param response
+     * @param out
+     * @param sm
+     * @param sess
+     * @param locale
+     * @param xpath
+     * @throws IOException
+     * @throws JSONException
+     *
+     * An earlier version of this code was formerly in RefreshRow.jsp.
+     * Moved to SurveyAjax.java to make more consistent with other requests, to reduce the amount of code needed,
+     * to enable features of Eclipse that don't work correctly for jsp files, and to facilitate refactoring
+     * getSection to improve performance related to DataSection.
+     * Reference: https://unicode-org.atlassian.net/projects/CLDR/issues/CLDR-11877
+     *        and https://unicode-org.atlassian.net/projects/CLDR/issues/CLDR-12020
+     */
+    public static void getRow(HttpServletRequest request, HttpServletResponse response, Writer out,
+            SurveyMain sm, String sess, CLDRLocale locale, String xpath)
+            throws IOException, JSONException {
+
+        CLDRConfigImpl.setUrls(request);
+        WebContext ctx = new WebContext(request, response);
+        ElapsedTimer et = new ElapsedTimer();
+
+        String loc = locale.toString();
+        ctx.setLocale(locale);
+        xpath = WebContext.decodeFieldString(xpath); // TODO: why doesn't processRequest do decodeFieldString? Not needed, all ASCII?
```

Comment:
Probably assumed to be ASCII

Reply:
OK. That does appear to be assumed elsewhere.

Reply:
may be a latent bug there.